operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java

```diff
@@ -93,6 +93,12 @@
      */
     public static final String KUBERNETES_STATEFULSET_POD_LABEL = "statefulset.kubernetes.io/pod-name";
 
+    /**
+     * Used to exclude parent CR's labels from being assigned to provisioned subresources
+     */
+    public static final String STRIMZI_LABELS_EXCLUSION_PATTERN = System.getenv()
+            .getOrDefault("STRIMZI_LABELS_EXCLUSION_PATTERN", "^app.kubernetes.io/(?!part-of).*");
```

Comment:
Have you done any tests to see that everything works fine when the Kube labels which we are excluded by default are copied over to other resources? Does everything seem to workM

Reply:
I did some testing locally, deployed strimzi operator using minikube and tried a few different regex patterns to see how kafka cluster labels would look, seems working as expected. For example: 
Operator deployment
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: strimzi-cluster-operator
  labels:
    app: strimzi
spec:
  replicas: 1
  selector:
    matchLabels:
      name: strimzi-cluster-operator
      strimzi.io/kind: cluster-operator
  template:
    metadata:
      labels:
        name: strimzi-cluster-operator
        strimzi.io/kind: cluster-operator
    spec:
      serviceAccountName: strimzi-cluster-operator
      volumes:
        - name: strimzi-tmp
          emptyDir:
            medium: Memory
        - name: co-config-volume
          configMap:
            name: strimzi-cluster-operator
      containers:
        - name: strimzi-cluster-operator
          image: kchernou/operator:latest
          imagePullPolicy: IfNotPresent
          ports:
            - containerPort: 8080
              name: http
          args:
            - /opt/strimzi/bin/cluster_operator_run.sh
          volumeMounts:
            - name: strimzi-tmp
              mountPath: /tmp
            - name: co-config-volume
              mountPath: /opt/strimzi/custom-config/
          env:
            - name: STRIMZI_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: STRIMZI_FULL_RECONCILIATION_INTERVAL_MS
              value: "120000"
            - name: STRIMZI_OPERATION_TIMEOUT_MS
              value: "300000"
            - name: STRIMZI_LABELS_EXCLUSION_PATTERN
              value: "^key1.*"
...
```
Kafka
```
Name:         my-cluster
Namespace:    myproject
Labels:       app.kubernetes.io/test=value
              key1=value
              key2=value
Annotations:  <none>
API Version:  kafka.strimzi.io/v1beta2
Kind:         Kafka
```
Zookeeper pod
```
Name:           my-cluster-zookeeper-0
Namespace:      myproject
Priority:       0
Node:           minikube/192.168.49.2
Start Time:     Mon, 26 Apr 2021 11:06:45 +1000
Labels:         app.kubernetes.io/instance=my-cluster
                app.kubernetes.io/managed-by=strimzi-cluster-operator
                app.kubernetes.io/name=zookeeper
                app.kubernetes.io/part-of=strimzi-my-cluster
                app.kubernetes.io/test=value
                controller-revision-hash=my-cluster-zookeeper-7ffbb7f7dd
                key2=value
                statefulset.kubernetes.io/pod-name=my-cluster-zookeeper-0
                strimzi.io/cluster=my-cluster
                strimzi.io/kind=Kafka
                strimzi.io/name=my-cluster-zookeeper
Annotations:    strimzi.io/cluster-ca-cert-generation: 0
                strimzi.io/generation: 0
                strimzi.io/logging-hash: 0f057cb0003c78f02978b83e4fabad5bd508680c
```
Kafka bootstrap service
```
Name:              my-cluster-kafka-bootstrap
Namespace:         myproject
Labels:            app.kubernetes.io/instance=my-cluster
                   app.kubernetes.io/managed-by=strimzi-cluster-operator
                   app.kubernetes.io/name=kafka
                   app.kubernetes.io/part-of=strimzi-my-cluster
                   app.kubernetes.io/test=value
                   key2=value
                   strimzi.io/cluster=my-cluster
                   strimzi.io/discovery=true
                   strimzi.io/kind=Kafka
                   strimzi.io/name=my-cluster-kafka
Annotations:       strimzi.io/discovery:
                     [ {
                       "port" : 9092,
```
Kafka pod with labels from template
```
Name:         my-cluster-kafka-0
Namespace:    myproject
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Mon, 26 Apr 2021 11:07:57 +1000
Labels:       app.kubernetes.io/instance=my-cluster
              app.kubernetes.io/managed-by=strimzi-cluster-operator
              app.kubernetes.io/name=kafka
              app.kubernetes.io/part-of=strimzi-my-cluster
              app.kubernetes.io/test=value
              controller-revision-hash=my-cluster-kafka-85bbd68f87
              key1=from-template
              key2=value
              statefulset.kubernetes.io/pod-name=my-cluster-kafka-0
              strimzi.io/cluster=my-cluster
              strimzi.io/kind=Kafka
              strimzi.io/name=my-cluster-kafka
Annotations:  strimzi.io/broker-configuration-hash: f471dd811e1566753da5da4a12a2d85575a3e132d68832ad309ffe2bd29616f73d99ab9217da3512
              strimzi.io/clients-ca-cert-generation: 0
```
Zookeeper pod after removing STRIMZI_LABELS_EXCLUSION_PATTERN from operator deployment to test defaults
```
Name:         my-cluster-zookeeper-0
Namespace:    myproject
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Mon, 26 Apr 2021 11:15:59 +1000
Labels:       app.kubernetes.io/instance=my-cluster
              app.kubernetes.io/managed-by=strimzi-cluster-operator
              app.kubernetes.io/name=zookeeper
              app.kubernetes.io/part-of=strimzi-my-cluster
              controller-revision-hash=my-cluster-zookeeper-847bd8d777
              key1=value
              key2=value
              statefulset.kubernetes.io/pod-name=my-cluster-zookeeper-0
              strimzi.io/cluster=my-cluster
              strimzi.io/kind=Kafka
              strimzi.io/name=my-cluster-zookeeper
Annotations:  strimzi.io/cluster-ca-cert-generation: 0
              strimzi.io/generation: 1
```

Reply:
What would happen for example when your `Kafka` CR was named `my-cluster` but had a labels `app.kubernetes.io/instance=something-else`?

Reply:
It will be app.kubernetes.io/instance=my-cluster even if the user provided label isn't filtered out as the  `generateDefaultLabels` method will call `.withKubernetesInstance` after processing user provided labels:
https://github.com/strimzi/strimzi-kafka-operator/blob/main/operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java#L425

Test results with `STRIMZI_LABELS_EXCLUSION_PATTERN=.*`
Kafka
```
Name:         my-cluster
Namespace:    myproject
Labels:       app.kubernetes.io/instance=something-else
              key1=value
              key2=value
Annotations:  <none>
API Version:  kafka.strimzi.io/v1beta2
Kind:         Kafka
```
Zookeeper pod
```
Name:         my-cluster-zookeeper-0
Namespace:    myproject
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Tue, 27 Apr 2021 13:13:15 +1000
Labels:       app.kubernetes.io/instance=my-cluster
              app.kubernetes.io/managed-by=strimzi-cluster-operator
              app.kubernetes.io/name=zookeeper
              app.kubernetes.io/part-of=strimzi-my-cluster
              controller-revision-hash=my-cluster-zookeeper-b788c75f9
              statefulset.kubernetes.io/pod-name=my-cluster-zookeeper-0
              strimzi.io/cluster=my-cluster
              strimzi.io/kind=Kafka
              strimzi.io/name=my-cluster-zookeeper
Annotations:  strimzi.io/cluster-ca-cert-generation: 0
              strimzi.io/generation: 0
              strimzi.io/logging-hash: 0f057cb0003c78f02978b83e4fabad5bd508680c
Status:       Running
IP:           172.17.0.4
```
Test with default STRIMZI_LABELS_EXCLUSION_PATTERN
Zookeeper pod
```
Name:         my-cluster-zookeeper-0
Namespace:    myproject
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Tue, 27 Apr 2021 13:27:21 +1000
Labels:       app.kubernetes.io/instance=my-cluster
              app.kubernetes.io/managed-by=strimzi-cluster-operator
              app.kubernetes.io/name=zookeeper
              app.kubernetes.io/part-of=strimzi-my-cluster
              controller-revision-hash=my-cluster-zookeeper-847bd8d777
              key1=value
              key2=value
              statefulset.kubernetes.io/pod-name=my-cluster-zookeeper-0
              strimzi.io/cluster=my-cluster
              strimzi.io/kind=Kafka
              strimzi.io/name=my-cluster-zookeeper
Annotations:  strimzi.io/cluster-ca-cert-generation: 0
              strimzi.io/generation: 1
              strimzi.io/logging-hash: 0f057cb0003c78f02978b83e4fabad5bd508680c
Status:       Running
```

Reply:
Zookeeper pod with `STRIMZI_LABELS_EXCLUSION_PATTERN="key1"`
```
Name:         my-cluster-zookeeper-0
Namespace:    myproject
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Tue, 27 Apr 2021 13:41:50 +1000
Labels:       app.kubernetes.io/instance=my-cluster
              app.kubernetes.io/managed-by=strimzi-cluster-operator
              app.kubernetes.io/name=zookeeper
              app.kubernetes.io/part-of=strimzi-my-cluster
              controller-revision-hash=my-cluster-zookeeper-75df4fd76c
              key2=value
              statefulset.kubernetes.io/pod-name=my-cluster-zookeeper-0
              strimzi.io/cluster=my-cluster
              strimzi.io/kind=Kafka
              strimzi.io/name=my-cluster-zookeeper
Annotations:  strimzi.io/cluster-ca-cert-generation: 0
              strimzi.io/generation: 2
              strimzi.io/logging-hash: 0f057cb0003c78f02978b83e4fabad5bd508680c
Status:       Running
IP:           172.17.0.4
```